delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphStreamBuilder.scala

```diff
@@ -0,0 +1,117 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.indexing
+
+import akka.http.scaladsl.model.Uri
+import cats.syntax.functor._
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.ClasspathResourceUtils
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.client.{BlazegraphClient, SparqlWriteQuery}
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.model.BlazegraphView.IndexingBlazegraphView
+import ch.epfl.bluebrain.nexus.delta.plugins.blazegraph.model.BlazegraphViewsConfig
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.RdfError.InvalidIri
+import ch.epfl.bluebrain.nexus.delta.rdf.graph.{Graph, NTriples}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution
+import ch.epfl.bluebrain.nexus.delta.sdk.ProgressesStatistics.ProgressesCache
+import ch.epfl.bluebrain.nexus.delta.sdk.implicits._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{BaseUri, ResourceF}
+import ch.epfl.bluebrain.nexus.delta.sdk.views.indexing.StreamBuilder
+import ch.epfl.bluebrain.nexus.delta.sdk.views.indexing.StreamBuilder.ProgressStrategy
+import ch.epfl.bluebrain.nexus.delta.sdk.views.model.ViewIndex
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.ProjectionProgress.NoProgress
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.ProjectionStream.{ChunkStreamOps, SimpleStreamOps}
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.{Projection, ProjectionId}
+import fs2.Stream
+import monix.bio.{IO, Task}
+import monix.execution.Scheduler
+
+/**
+  * Defines how to build a stream for an IndexingBlazegraphView
+  */
+final class BlazegraphStreamBuilder(
+    client: BlazegraphClient,
+    cache: ProgressesCache,
+    config: BlazegraphViewsConfig,
+    eventLog: BlazegraphIndexingEventLog,
+    projection: Projection[Unit]
+)(implicit cr: RemoteContextResolution, baseUri: BaseUri, sc: Scheduler)
+    extends StreamBuilder[IndexingBlazegraphView] {
+  override def apply(
+      prev: Option[ViewIndex[IndexingBlazegraphView]],
+      view: ViewIndex[IndexingBlazegraphView],
+      progressStrategy: StreamBuilder.ProgressStrategy
+  ): Stream[Task, Unit] = {
+
+    implicit val projectionId: ProjectionId = view.projectionId
+    Stream
+      .eval {
+        for {
+          _        <- prev.fold(Task.unit)(prev => client.deleteNamespace(prev.index).void >> cache.remove(prev.projectionId))
+          props    <- ClasspathResourceUtils.ioPropertiesOf("blazegraph/index.properties")
+          _        <- client.createNamespace(view.index, props)
```

Comment:
We don't have to attempt to create the namespace everytime, just if it is a new view or if it has been updated

Reply:
Even though this is true, this operation is not expensive. To add the logic to make this work is probably worse than just try every time we build a stream.

Reply:
In the current implementation, it is a:
```
IO.when(prev.isDefined) { client.createNamespace(view.index, props)  }
```

Reply:
that's wrong. The first time you don't have a previous

Reply:
My bad, I need another coffee I guess :).
But the point was it should not be that complicated

Reply:
Yes it shouldn't. But my point is that it is not worth it, since those methods already handle well the cases when the index/namespace already exists.

In this case I prefer a consistent and simple logic, than to pass along more information just to fulfill this.